app/controllers/company.controller.js

```diff
@@ -0,0 +1,130 @@
+const db = require('../models');
+
+const {
+  Company,
+  User,
+  Sequelize: { Op },
+} = db;
+
+// Create a company
+exports.createCompany = async (req, res, next) => {
+  const {
+    body: {
+      userId,
+      companyName,
+      website,
+      primaryColor,
+      secondaryColor,
+      address,
+      logo,
+      bio,
+    },
+  } = req;
+
+  User.findById(userId)
+    .then((user) => {
+      user.update({ isCompanyOwner: true });
+      return Company.findOrCreate({
+        where: {
+          [Op.or]: [{ companyName }, { website }],
+        },
+        defaults: {
+          companyName,
+          website,
+          ownerUserId: userId,
+          primaryColor: primaryColor || '#000000',
+          secondaryColor: secondaryColor || '#000000',
+          address: address || '',
+          logo: logo || 'https://pbs.twimg.com/media/FB6YhR8WQAI5MnM.png',
+          bio: bio || '',
+          createdAt: Date.now(),
+        },
+      });
+    })
+    .then(([company, created]) => {
+      if (!created) {
+        return res.status(400).send({
+          message: 'Company already exists',
+        });
+      }
```

Comment:
needs a transaction. what if company already exists? user will have `isCompanyOwner == true` but the company won't be created